source-postgres/replication.go

```diff
@@ -420,10 +422,14 @@ func (s *replicationStream) decodeChangeEvent(
 		return nil, fmt.Errorf("error encoding row key for %q: %w", streamID, err)
 	}
 
-	if err := translateRecordFields(nil, bf); err != nil {
+	discovery, ok := s.tables.discovery[streamID]
+	if !ok {
+		return nil, fmt.Errorf("unknown discovery info for stream %q", streamID)
+	}
+	if err := translateRecordFields(discovery, bf); err != nil {
```

Comment:
I found a bug where during replication we were not providing discovery information to `translateRecordFields` (this was not noticed because we did not rely on discovery info in this function up to this point)